회원가입 Form 작성 페이지 UI 작성

riverandeye opened the pull request:
신규 회원가입 시 추가 정보를 제출하는 페이지를 작성하였습니다.
Nickname과 체크박스가 입력되지 않을 시 버튼이 활성화되지 않는 기능을 추가하였습니다.

riverandeye committed:
[client] | feat(LoginForm): 회원가입 Form 작성 페이지 생성

신규 회원가입 시 추가 정보를 제출하는 페이지를 작성하였습니다.
Nickname과 체크박스가 입력되지 않을 시 버튼이 활성화되지 않는 기능을 추가하였습니다.

hyokualexkwon commented on packages/client/pages/loginform.tsx:
```diff
@@ -0,0 +1 @@
+export { default } from '../views/LoginForm';
```
사용자는 `/loginform`이라는 경로에 접근하게 될텐데, path가 semantic하다고 생각되지 않습니다. 

이제보니 인증페이지도 `/login` 경로에서 접근하게 되는데, 회원가입 기능도 한번에 하는 페이지임을 고려하면 충분히 의미를 담고 있는것 같지 않습니다.

차라리 인증 페이지는 `/auth`, 그리고 회원가입 양식 페이지는 `/auth/additional-info` 와 같은 path로 표현하는건 어떨까요?

riverandeye replied:
정말 좋은 의견이라고 생각합니다. 수렴하겠습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+import * as S from './styles';
+
+const LoginForm: React.FunctionComponent = () => {
+  const [input, setInput] = useState({
```
form의 데이터를 다루는 state같은데, input이라는 이름이 사용되고 있네요. form은 input만을 포함하는게 아닐텐데 조금 더 포괄적인 이름이 없을까요?

riverandeye replied:
formData로 가겠습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+  const [input, setInput] = useState({
+    nickname: '',
+    agree: false,
+  });
```
소개 항목에 작성한 데이터는 어디서 관리되고 있나요?

riverandeye replied:
반영했습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+          </S.Input>
+          <S.Input>
+            <S.InputInfo>
+              <label htmlFor="introduction">소개</label>
```
db entity에 따르면, 유저 소개는 description이라는 이름으로 사용되고 있는데, 같은 이름을 쓰는게 더 좋지 않을까요?

riverandeye replied:
반영하였습니다

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+        <S.Form onSubmit={() => null}>
+          <S.Input>
+            <S.InputInfo>
+              <label htmlFor="nickname">
```
db entity에서 사용자 닉네임은 username이라는 이름으로 사용하고 있는데, 같은 이름을 쓰는게 더 좋지 않을까요?

riverandeye replied:
반영했습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+          <S.Input>
+            <S.InputInfo>
+              <label htmlFor="nickname">
+                닉네임<S.RequireMark></S.RequireMark>
```
RequireMark는 children을 넘기지 않는다면 self closing tag를 쓰는게 더 좋지 않을까요? `<RequireMark/>

riverandeye replied:
좋은 의견이에요! 수렴하겠습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+    agree: false,
+  });
+
+  const handleNickname = e => {
```
저는 이벤트 핸들러에는 이벤트 이름이 들어가면 좋다는 생각을 하고 있는데요, 예를 들어 handleUsernameChange와 같이 말이죠. 왜냐하면 Username에 또 다른 이벤트를 감지하게 되면 추가로 핸들러가 생길 수 있잖아요? handleUsernameClick과 같이 말이죠. 한번 고려해주시면 감사하겠습니다~

riverandeye replied:
좋은 의견이에요! 수렴하겠습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+              onChange={handleNickname}
+              type={'text'}
+              id="nickname"
+            ></input>
```
self closing tag!

riverandeye replied:
ㅎㅎㅎ수렴하겠습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+            ></input>
+          </S.Input>
+          <S.Input>
+            <S.InputInfo>
```
InputInfo라는 이름을 사용하셨는데, textarea를 설명하게 된다면 TextareaInfo가 되어야 하지 않을까요? 범용적인 이름을 사용해야 할 것 같습니다. label 정보를 담고 있으니 `Label`이라고 명명하는건 어떨까요?

riverandeye replied:
좋은 의견 감사합니다~ 반영했습니다!

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+              id="nickname"
+            ></input>
+          </S.Input>
+          <S.Input>
```
Input이라는 이름을 사용하셨는데, textarea를 설명하게 된다면 Textarea가 되어야 하지 않을까요? 범용적인 이름을 사용해야 할것 같습니다. form에 포함되어있는 입력 항목들을 의미한다면`Item`이라고 명명하는건 어떨까요?

riverandeye replied:
역시 좋은 생각이에요 ㅎㅎ

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+          </S.Input>
+          <S.Input>
+            <S.InputInfo>
+              <input onClick={handlecheck} type="checkbox" id="agree"></input>
```
self closing tag

riverandeye replied:
ㅎㅎ수렴하겠습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+          </S.Input>
+          <S.Input>
+            <S.InputInfo>
+              <input onClick={handlecheck} type="checkbox" id="agree"></input>
```
handlecheck? 이름만 봐서 무엇을 체크하는지 알 수 있나요? checkbox 또는 agree라는 이름이 들어가야하지 않을까요?

그리고 왜 여기는 camelcase가 아닌가요?

riverandeye replied:
반영하였습니다~

hyokualexkwon commented on packages/client/views/LoginForm/index.tsx:
```diff
@@ -0,0 +1,62 @@
+            </S.InputInfo>
+          </S.Input>
+          <S.SubmitButton>
+            <button disabled={active}>가입하기</button>
```
양식 제출 버튼의 disabled 여부가 단순히 양식이 채워졌는지 여부로만 검사하고있는데, 양식이 제출되어서 서버에 요청이 보내고 응답이 오기 전까지 disabled 하는 것은 어떻게 처리해야할까요?

riverandeye replied:
수렴하였습니다~

KimJye reviewed (approved):
확인했습니다

riverandeye committed:
[client] | refactor(login): 코드 컨벤션 및 수정사항 반영

1. 라우트를 변경하였습니다.
- /login -> /auth
- /loginform -> /auth/signup

2. 컴포넌트명을 요구사항에 맞게 변경하였습니다.

3. 직접 입력된 상수들을 constant로 분리하였습니다.
4. 적절한 변수명을 사용하도록 변경하였습니다.

Atercatus commented on packages/client/constants.ts:
```diff
@@ -22,3 +22,8 @@ export const fontWeight = {
+
+export const maxLength = {
+  username: 30,
+  introduction: 1500,
```
entity 에서는 3000인데 여기선 1500 인건 한글을 생각해서 인가여

riverandeye replied:
네 맞아요~ 글자 제한 자체도 1500자로 두어서 그렇습니다.

Atercatus commented on packages/client/views/Auth/index.tsx:
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import { LogoSVG, GitHubSVG } from '../../svgs';
 
-const Login: React.FunctionComponent = () => {
+const Auth: React.FunctionComponent = () => {
```
아래에는 `AuthenticateLink`라고 상세히 묘사 했지만 여기선 `Auth` 라고 하신 이유가 있나요? 
좀 더 상세한 의미가 좋을 거같은데 어떻게 생각하세요?

riverandeye replied:
폴더 이름이랑 똑같이 설정하려고 그렇게 했습니다.

Atercatus commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+import * as S from './styles';
+import { maxLength } from '../../constants';
+
+const AuthForm: React.FunctionComponent = () => {
```
`SignUp` 페이지에 있는 `AuthForm`은 의미가 명확하지 않네요.
`SignUpForm`을 말씀하시는 건가요

riverandeye replied:
네네 반영하였습니다.

Atercatus commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+import { maxLength } from '../../constants';
+
+const AuthForm: React.FunctionComponent = () => {
+  const [formData, setInput] = useState({
```
상태명은 `formData` 인데 그에 따른 함수명은 `setInput`이네요. 
의미상으로 좀 더 통일성을 가지면 좋을거 같습니다.

hyokualexkwon replied:
이 부분은 리액트 관습에 따르면, formData => setFormData 라고 지으면 좋을것 같네여~

riverandeye replied:
반영하였습니다

Atercatus commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+  };
+
+  const handleCheckboxChange = e => {
+    setInput({ ...formData, agree: !formData.agree });
```
의미상으로 너무 난해합니다.
`agree`에 `!formData.agree` 을 대입하는 표현은 설명을 하지않으면 도무지 해석이 불가능합니다.

riverandeye replied:
반영하였습니다~

Atercatus commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+    agree: false,
+  });
+
+  const [sending, setSending] = useState(false);
```
다른 표현이 좋을 거 같습니다. boolean 을 `isXXX`로 네이밍하기로 하셨다면 통일을 하거나 좀 더 명확한 표현이 좋겠네요

riverandeye replied:
반영하였습니다~

Atercatus commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+  };
+
+  const isFormFilled = !formData.agree || formData.username === '';
+  const isButtonDisabled = sending ? true : isFormFilled;
```
짧게 만들기 위해 가독성을 포기해버린 코드입니다.
짧은 코드가 가독성이 높은 코드가 아니라 그냥 읽었을 때 의도가 보이는 코드가 좋은 코드입니다.  조건문에 대해서 삼항 연산자를 겹쳐 쓰면 종종 이런 의도가 보이지 않는 코드가 발생합니다. 

```javascript
function checkFormVerified() {
  return !(formData.isAgree && formData.username);
}
const isButtonDisabled = checkFormVerified() && isSending;
```
 와 같은 방법을 이용해 주세요

riverandeye replied:
반영하였습니다~

Atercatus reviewed (changes requested):
서비스약관 추가와, 가입하기 버튼에 cursor 속성을 변경해야 할 거 같습니다.

hyokualexkwon commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+import * as S from './styles';
+import { maxLength } from '../../constants';
+
+const AuthForm: React.FunctionComponent = () => {
```
컴포넌트 이름이 파일 디렉토리 이름을 따라가면 좋을것 같아요! `SignUp`과 같이 말이죠~

riverandeye replied:
반영하였습니다~

hyokualexkwon commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,90 @@
+            <input
+              maxLength={maxLength.username}
+              onChange={handleUsernameChange}
+              type={'text'}
```
type="text" 와 같이 작성하는게 더 자연스러울것 같아요!

riverandeye replied:
반영하였습니다~

riverandeye committed:
[client] | refactor(auth): PR 및 수정 요구사항 반영

1. login 페이지 url을 /auth -> /auth/login으로 변경
2. signup 페이지 url을 /signup으로 설정
3. login 페이지와 signup 페이지 마크업을 마우스로 드래그하지 못하게 수정함.
4. 기타 camelCase 및 함수 리팩토링

hyokualexkwon commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,89 @@
+    setFormData({ ...formData, username: e.target.value });
+  };
+
+  const handleIntroduction = e => {
```
왜 description의 핸들러가 handleIntroduction인가요? 같은 네이밍을 사용하는것이 덜 헷갈리지 않을까요?

hyokualexkwon commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,89 @@
+        <S.Form onSubmit={() => null}>
+          <S.Item>
+            <S.Label>
+              <label htmlFor="username">
```
htmlFor를 사용하셨는데, 가리키는 input이 없어 보이는데요? for가 가르키는것은 어떤 input의 id를 가리키는 것으로 알고 있어요. 닉네임을 작성하는 input에 id="username"을 작성하셔야할것 같아요

riverandeye replied:
리팩토링하는 과중에 지워버렸네요 다시 반영하겠습니다.

hyokualexkwon commented on packages/client/views/SignUp/index.tsx:
```diff
@@ -0,0 +1,89 @@
+
+  const [isFetching, setIsFetching] = useState(false);
+
+  const handleUsername = e => {
```
input 과 textarea 마다 핸들러를 별도로 작성하셨는데, name을 활용해서 조금 더 간결하게 쓸 수 있어요

```
const handleChange = e => {
  const { name, value } = e.target;
  setFormData({
    ...formData,
    [name]: value
  });
}
```

이런 방법은 어떨까요?

riverandeye replied:
오오 한번 시도해보겠습니다.

riverandeye committed:
[client] | refactor(auth): PR 및 수정 요구사항 반영

1. label for에 맞는 input의 id를 추가하였습니다.
2. 핸들러 함수를 하나로 통합하였습니다.

hyokualexkwon reviewed (approved):
고생 많으셨네요!  👍

riverandeye commented:
> 고생 많으셨네요! 👍

덕분에 많이 배웁니다 ㅠ.ㅠ
